File: src/main/java/com/likelionknu/applyserver/application/data/dto/request/FinalSubmitRequestDto.java
package com.likelionknu.applyserver.application.data.dto.request;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record FinalSubmitRequestDto(
        @NotEmpty @Valid
        List<Item> items
) {
    public record Item(
            @NotNull Long questionId,
            @NotBlank String answer
    ) {}
}

File: src/main/kotlin/com/likelionknu/applyserver/application/data/dto/request/FinalSubmitRequestDto.kt
package com.likelionknu.applyserver.application.data.dto.request

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull

data class FinalSubmitRequestDto(
    @field:NotNull
    val recruitId: Long,

    @field:NotEmpty
    @field:Valid
    val items: List<Item>
) {
    data class Item(
        @field:NotNull
        val questionId: Long,

        @field:NotBlank
        val answer: String
    )
}